### ISSUE TITLE
Spectrogram code review

### ISSUE THREAD

**Issue by f0k** [keunwoochoi/torchaudio-contrib#22]
Not sure if we are at this level of detail yet, but maybe it can serve as general hints for the code:
- The `Spectrogram` takes an `sr` argument, but never uses it. It should not know about the sample rate.
- On a related note, it defines a `_out_seq_dim()` method that's never used (and does not take into account the padding).
- It does not match the API described in the README, which allows to define the padding mode.
- It does not match the functional API, which does not allow padding at all. Either both or none should support padding. If both support padding, it should be handled in the functional implementation (which should leave it up to `torch.stft()`, since that one might eventually switch to a more efficient implementation than an extra `torch.nn.functional.pad` call).
- Both the constructor and `_build_window` handle the `None` case. Only one should. On a related note, I'm not sure the separate `_build_window` method is needed, is this ever going to be called outside the constructor? It's not so much code that it pays off to pull it out, in my opinion; it rather makes the constructor harder to read.
- The docstring format does not follow the conventions in Pytorch.

**Comment by keunwoochoi** [comment]
Apparently you're far ahead the current state of the code, which is good :)
- `sr` should be removed.
- `_out_seq_dim()`: forwarding it to @ksanjeevan 
- Right, neither the default values. I think it should follow the API in READMe.
- Agreed on both - it should follow the functional, additionally because I think implementations of `Layer` should re-use the implementation of `Functional` in general.
- `_build_window` - it seems worth discussing. I'm inclined to have an explicit default parameter like `"hann"` in the constructor and use it during initialization.
- Right, we'll need to handle it when we're ready to PR to torch/audio.

**Comment by ksanjeevan** [comment]
`_out_seq_dim()` should be removed yes.
